drivers/sensor/lis2de12/lis2de12.c

```diff
@@ -0,0 +1,479 @@
+#endif
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static inline void lis2de12_accel_convert(struct sensor_value *val, int raw_val,
+					  uint32_t sensitivity)
+{
+	int64_t dval;
+
+	/* Sensitivity is exposed in ug/LSB */
+	/* Convert to m/s^2 */
+	dval = (int64_t)(raw_val / 256) * sensitivity * SENSOR_G_DOUBLE;
+	val->val1 = (int32_t)(dval / 1000000);
+	val->val2 = (int32_t)(dval % 1000000);
+
+}
+
+static inline int lis2de12_accel_get_channel(enum sensor_channel chan,
+					     struct sensor_value *val,
+					     struct lis2de12_data *data,
+					     uint32_t sensitivity)
+{
+	uint8_t i;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		lis2de12_accel_convert(val, data->acc[0], sensitivity);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		lis2de12_accel_convert(val, data->acc[1], sensitivity);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		lis2de12_accel_convert(val, data->acc[2], sensitivity);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		for (i = 0; i < 3; i++) {
+			lis2de12_accel_convert(val++, data->acc[i], sensitivity);
+		}
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lis2de12_accel_channel_get(enum sensor_channel chan,
+				      struct sensor_value *val,
+				      struct lis2de12_data *data)
+{
+	return lis2de12_accel_get_channel(chan, val, data, data->acc_gain);
+}
+
+#if defined(CONFIG_LIS2DE12_ENABLE_TEMP)
+static void lis2de12_temp_channel_get(struct sensor_value *val, struct lis2de12_data *data)
+{
+	int64_t micro_c;
+
+	/* convert units to micro Celsius. Raw temperature samples are
+	 * expressed in 256 LSB/deg_C units. And LSB output is 0 at 25 C.
+	 */
+	micro_c = ((int64_t)data->temp_sample * 1000000) / 256;
+
+	val->val1 = micro_c / 1000000 + 25;
+	val->val2 = micro_c % 1000000;
+}
+#endif
+
+static int lis2de12_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct lis2de12_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		lis2de12_accel_channel_get(chan, val, data);
+		break;
+#if defined(CONFIG_LIS2DE12_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		lis2de12_temp_channel_get(val, data);
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api lis2de12_driver_api = {
+	.attr_set = lis2de12_attr_set,
+#if CONFIG_LIS2DE12_TRIGGER
+	.trigger_set = lis2de12_trigger_set,
+#endif
+	.sample_fetch = lis2de12_sample_fetch,
+	.channel_get = lis2de12_channel_get,
+};
+
+static int lis2de12_init_chip(const struct device *dev)
+{
+	const struct lis2de12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	struct lis2de12_data *lis2de12 = dev->data;
+	uint8_t chip_id;
+	uint8_t odr, fs;
+
+	if (lis2de12_device_id_get(ctx, &chip_id) < 0) {
+		LOG_ERR("Failed reading chip id");
+		return -EIO;
+	}
+
+	LOG_INF("chip id 0x%x", chip_id);
```

Comment:
I would put this after the chip ID check so you either get the info OR the error

Reply:
sure!